src/main/scala/subsystem/HasTiles.scala

```diff
@@ -223,98 +226,121 @@ trait CanAttachTile {
 
     //    From CLINT: "msip" and "mtip"
-    tile.crossIntIn() :=
+    domain.crossIntIn(crossingParams.crossingType) :=
       context.clintOpt.map { _.intnode }
         .getOrElse { NullIntSource(sources = CLINTConsts.ints) }
 
     //    From PLIC: "meip"
-    tile.crossIntIn() :=
+    domain.crossIntIn(crossingParams.crossingType) :=
       context.plicOpt .map { _.intnode }
         .getOrElse { context.meipNode.get }
 
     //    From PLIC: "seip" (only if supervisor mode is enabled)
-    if (tile.tileParams.core.hasSupervisorMode) {
-      tile.crossIntIn() :=
+    if (domain.tile.tileParams.core.hasSupervisorMode) {
+      domain.crossIntIn(crossingParams.crossingType) :=
         context.plicOpt .map { _.intnode }
           .getOrElse { NullIntSource() }
     }
 
     // 3. Local Interrupts ("lip") are required to already be synchronous to the Tile's clock.
-    // (they are connected to tile.intInwardNode in a seperate trait)
+    // (they are connected to domain.tile.intInwardNode in a seperate trait)
 
     // 4. Interrupts coming out of the tile are sent to the PLIC,
     //    so might need to be synchronized depending on the Tile's crossing type.
     context.plicOpt.foreach { plic =>
       FlipRendering { implicit p =>
-        plic.intnode :=* tile.crossIntOut()
+        plic.intnode :=* domain.crossIntOut(crossingParams.crossingType)
       }
     }
+  }
 
-    // 5. Notifications of tile status are collected without needing to be clock-crossed
+  /** Notifications of tile status are connected to be broadcast without needing to be clock-crossed. */
+  def connectOutputNotifications(tile: TileType, context: TileContextType): Unit = {
+    implicit val p = context.p
     context.tileHaltXbarNode :=* tile.haltNode
     context.tileWFIXbarNode :=* tile.wfiNode
     context.tileCeaseXbarNode :=* tile.ceaseNode
   }
 
-  /** Connect the input to the tile that are assumed to be constant during normal operation. */
+  /** Connect inputs to the tile that are assumed to be constant during normal operation, and so are not clock-crossed. */
   def connectInputConstants(tile: TileType, context: TileContextType): Unit = {
     implicit val p = context.p
+    val tlBusToGetPrefixFrom = context.locateTLBusWrapper(crossingParams.mmioBaseAddressPrefixWhere)
     tile.hartIdNode := context.tileHartIdNode
     tile.resetVectorNode := context.tileResetVectorNode
-    context.locateTLBusWrapper(crossingParams.mmioBaseAddressPrefixWhere).prefixNode.foreach {
-      tile.mmioAddressPrefixNode := _
-    }
+    tlBusToGetPrefixFrom.prefixNode.foreach { tile.mmioAddressPrefixNode := _ }
+  }
+
+  /** Connect power/reset/clock resources. */
+  def connectPRC(domain: TilePRCIDomain[TileType], context: TileContextType): Unit = {
+    implicit val p = context.p
+    val tlBusToGetClockDriverFrom = context.locateTLBusWrapper(crossingParams.master.where)
+    domain.clockNode :=
+      crossingParams.injectClockNode(context) :=
```

Comment:
I think the answer is "yes" based on my read but want to confirm: this supports injecting unique sequences of clock nodes per tile, right?

Reply:
Also, hierarchically where do the injected clock nodes reside?

Reply:
Yes, and new wrapper that is hierarchically now the parent of each tile.